### DISCUSSION TITLE
How do I know my lib works with the new architecture?

### DISCUSSION THREAD

**Discussion by gabriel-logan** [reactwg/react-native-new-architecture#224] [Libraries]
I developed a library and I want to know for sure if it is fully compatible with the new architecture. I created it using create-react-native-library without using turbo or fabric. I tested it in an app using the new architecture and it worked. But can I be sure that it works with the new architecture?

So I can update it in the community directories?

Mylib: https://github.com/gabriel-logan/mobile-native-barcode-generator

**Comment by cortinico** [comment]
> I tested it in an app using the new architecture and it worked. But can I be sure that it works with the new architecture?

If you tested your library and is fully working, you can safely update the Directory.

**Reply by gabriel-logan** [reply]
> > I tested it in an app using the new architecture and it worked. But can I be sure that it works with the new architecture?
> 
> If you tested your library and is fully working, you can safely update the Directory.

I tested it only in dev mode, is it guaranteed that everything will work ok with the final release build? aab and apk ??

**Reply by cortinico** [reply]
> is it guaranteed that everything will work ok with the final release build? aab and apk ??

It's suggested you test in both debug and release. We don't have an exhaustive/automatic way to check if a library works with all the of the features, you will have to test manually

**Comment by gabriel-logan** [comment]
One thing about my tests, using the old architecture, the colors look ok in the texts and background, I'm not adding any colors, I'm leaving the default ones.

However, using the new architecture, the texts are black, even though I'm in dark mode.

And in this case the texts should be white and the background dark.
